deploy.sh

```
@@ -16,35 +16,137 @@ check_docker_daemon() {
-# Function to download and run docker-compose
+# Function to create or validate the .env file
+create_or_validate_env_file() {
+  env_file=".env"
+
+  if [ ! -f "$env_file" ]; then
+    echo "$env_file does not exist. Creating it now..."
+    touch "$env_file"
+  fi
+
+  # Source existing .env file to retain values
+  [ -f "$env_file" ] && source "$env_file"
```

Shellcheck (0.10.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)